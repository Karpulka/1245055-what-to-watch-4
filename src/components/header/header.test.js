import React from "react";
import renderer from "react-test-renderer";
import Header from "./header";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";

const mockStore = configureStore([]);

it(`Render Header Movie isAuth`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`
    }
  });

  const tree = renderer
    .create(<Provider store={store}><Header pageType={`MOVIE`}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Header Movie NoIsAuth`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`
    }
  });

  const tree = renderer
    .create(<Provider store={store}><Header pageType={`MOVIE`}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Header Sign In NoIsAuth`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`
    }
  });

  const tree = renderer
    .create(<Provider store={store}><Header pageType={`AUTH`}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Header Sign In Auth`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`
    }
  });

  const tree = renderer
    .create(<Provider store={store}><Header pageType={`AUTH`}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
