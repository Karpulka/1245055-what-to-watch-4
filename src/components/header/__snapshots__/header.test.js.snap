// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Movie NoIsAuth 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
  >
    Sign in
  </a>
</div>
`;

exports[`Render Header Movie isAuth 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`Render Header Sign In Auth 1`] = `
<h1
  className="page-title user-page__title"
>
  Sign in
</h1>
`;

exports[`Render Header Sign In NoIsAuth 1`] = `
<h1
  className="page-title user-page__title"
>
  Sign in
</h1>
`;
